controllers/user.go

```diff
@@ -1,31 +1,112 @@
+
+	"github.com/gin-gonic/gin"
+	"golang.org/x/crypto/bcrypt"
 )
 
-func GetUsers(c *gin.Context) {
-  var users []models.User
-  db.DB.Find(&users)
-  c.IndentedJSON(http.StatusOK, gin.H{"data": users})
+//UserController ...
+type UserController struct{}
+
+func (ctrl UserController) RegisterUser(c *gin.Context) {
+	var input models.RegisterInput
+	err := c.ShouldBind(&input)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error(), "success": false})
+		return
+	}
+
+	var users models.User
+	db.DB.Find(&users, "email=?", input.Email)
+
+	if users.Email == input.Email {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": "Email is Already Registered", "success": false})
+		return
+	}
+
+	bytePassword := []byte(input.Password)
+	hashedPassword, err := bcrypt.GenerateFromPassword(bytePassword, bcrypt.DefaultCost)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error(), "success": false})
+		return
+	}
+
+	// Create new user
+	user := models.User{Name: input.Name, Email: input.Email, Password: string(hashedPassword)}
+
+	db.DB.Create(&user)
+
+	c.IndentedJSON(http.StatusOK, gin.H{"success": true})
+}
+
+func (ctrl UserController) LoginUser(c *gin.Context) {
+
+	var input models.LoginInput
+	err := c.ShouldBind(&input)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error(), "success": false})
+		return
+	}
+
+	var user models.User
+	if db.DB.Find(&user, "email=?", input.Email).RecordNotFound() {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": "You have not registered, please register first", "success": false})
+		return
+	}
+
+	//Compare the password form and database if match
+	bytePassword := []byte(input.Password)
+	byteHashedPassword := []byte(user.Password)
+	err = bcrypt.CompareHashAndPassword(byteHashedPassword, bytePassword)
+
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusUnauthorized, gin.H{"error": "Incorrect Password", "success": false})
+		return
+	}
+	token, err := util.CreateToken(input.Email)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusUnauthorized, gin.H{"error": err.Error(), "success": false})
+		return
+	}
+
+	c.JSON(http.StatusOK, gin.H{"token": token, "success": true})
+}
+
+func (ctrl UserController) GetAll(c *gin.Context) {
+	var users []models.User
+	db.DB.Find(&users)
+
+	c.IndentedJSON(http.StatusOK, gin.H{"data": users, "success": true})
 }
 
-func CreateUser(c *gin.Context) {
-  // Validate input
-  var input models.UserInput
-  err := c.ShouldBindJSON(&input)
-  if err != nil {
-    c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
-    return
-  }
+func (ctrl UserController) GetOne(c *gin.Context) {
+	id := c.Param("id")
+
+	getID, err := strconv.ParseInt(id, 10, 64)
+	if getID == 0 || err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": "Invalid Parameter", "success": false})
+		return
+	}
```

Comment:
Wouldnt  be good to extract lines from 92:96 into a utility function?
Since you use in many places :)